### ISSUE TITLE
Beam elements in FEBio?

### ISSUE THREAD

**Issue by svizec** [febiosoftware/FEBio#48]
It would be very beneficial for some applications to have an option to use beam elements. The beam profile could just be circular, as that would probably account for most applications (surfaces can already be used for rectangular profiles). Is this somewhere on the roadmap by any chance?

**Comment by FEA-eng** [comment]
Indeed, beam/truss elements could represent ligaments or tendons, for instance.

**Comment by SteveMaas1978** [comment]
A rudimentary beam formulation has been implemented in FEBio. It is based on the formulation by Simo and Vu-Quoc,
https://doi.org/10.1016/0045-7825(88)90073-4

Here is an example file that should work with the current develop version. 
[cantilever.zip](https://github.com/febiosoftware/FEBio/files/13504009/cantilever.zip)

Please keep in mind that it is very likely that the implementation and the interface will change over next few weeks and months and we continue this development. 
Let us know if you have any questions or find any issues.

**Comment by svizec** [comment]
Thank you for the answer and for developing the software.

I tried running the supplied .feb file with FEVio 4.4.0, and it produced the following error:  **tag "material" (line 5) : invalid value for attribute "type"**.

I also tried opening the file in FEBioStudio 2.4, and it failed with a similar error.

**Comment by SteveMaas1978** [comment]
Make sure to install the development version. (The latest release version does not have this new feature yet.) 

See [here ](https://febio.org/knowledgebase/getting-started/updating-febio-and-febio-studio/) for instructions on how to get the latest development update.

**Comment by svizec** [comment]
> Make sure to install the development version. (The latest release version does not have this new feature yet.)
> 
> See [here ](https://febio.org/knowledgebase/getting-started/updating-febio-and-febio-studio/) for instructions on how to get the latest development update.

I realized that as soon as I replied to the original comment. It works as expected with the CLI solver and it completes the simulation, but the .feb file cannot be imported into FEBio Studio (Dev Version 2.4.0.a0387dfe2). 

The results can be visualized in FEBio Studio. Sphere-capped tubes when selecting "Render beams as solids" look a bit strange IMHO when the beam elements do not form a loop (both free ends could be flat faces). I assume this is due to how beams are rendered as solids in VTK. This is just a minor detail I noticed when looking at the results. 

Great work implementing the beam elements inside FEBio. I am looking forward to seeing how it develops further.

**Comment by SteveMaas1978** [comment]
Looks like I forgot to add support for reading quadratic beam elements in FEBio Studio. I have that fixed, and the file should load correctly in FBS. (For Windows, you can do a dev update now, for linux and mac tomorrow.) 

The beam visualization in FBS is indeed rather crude at the moment. I hope to add options to define beam profiles and then render the beam using the proper profile. For now, I've changed the rendering so that the beam ends are flat. 

Let me know if you find any other issues.

**Comment by svizec** [comment]
Are there any updates or an intention to include beam elements in contact formulations? (Beam-to-beam and beam-to-surface contacts).

Additionally, the contact algorithms currently only support choosing element faces for surfaces, even for n2f contacts. If I understand correctly, the nodal normal projections from the primary surface are calculated onto the facets of the secondary surface, and the primary surface is required to calculate nodal normals. But for tied contact interfaces, where only a specific edge (not a surface) has to be tied to another surface, the true node-to-surface (facet) definition (node set as primary "surface") would be a desirable solution. Such node-to-facet definition would also allow for an approximation of a sliding contact formulation between beam elements and surfaces.

Any plans of adding such functionalities that would help with endovascular device simulations? 

Also, adding an option to define boundary conditions and loads in cylindrical coordinates would be very interesting.

Thank you very much for actively developing great open-source software.
